### PR TITLE
chore: fix handling of metrics with created suffix

### PR DESCRIPTION
**Pull request by jmichalek132**
Description:
As discussed in the OTEL Prometheus working group, to fix this case we check If the metricfamily (which we get from the metadata) name + _created == the series name match. Then we treat it as created timestamp.
Link to tracking Issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30309

Testing: Reproduced based on the details provided in the issue, added a test case covering the case.

Documentation:

### CODE REVIEW

File: receiver/prometheusreceiver/internal/metricfamily.go [hunk 1]
```diff
@@ -297,7 +297,7 @@ func (mf *metricFamily) addSeries(seriesRef uint64, metricName string, ls labels
 			mg.complexValue = append(mg.complexValue, &dataPoint{value: v, boundary: boundary})
 		}
 	case pmetric.MetricTypeSum:
-		if strings.HasSuffix(metricName, metricSuffixCreated) {
+		if metricName == mf.metadata.Metric+metricSuffixCreated {
```
[the remark this answers is not in the document]
Comment (jmichalek132) [reply receiver/prometheusreceiver/internal/metricfamily.go:300]: Fixed. in last commit https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30808/commits/ba05905ab409957202537e38d2a62c485da989a8.

**Review by dashpole — COMMENTED** [review]
do we have tests that check that _created series are correctly parsed for summaries, histograms, and counters?

**Comment by jmichalek132** [comment]
> do we have tests that check that _created series are correctly parsed for summaries, histograms, and counters?

I added a test case for counters, will add one for summaries and histograms too.

**Comment by jmichalek132** [comment]
> do we have tests that check that _created series are correctly parsed for summaries, histograms, and counters?

Okay, so double checked there are positive tests for  _created when it's actually  _created timestamp being parsed and used for all of the three. We have also a test case for a counter where  _created is not the actual created timestamp and is being left as is. Should I add a test case for the latter for summaries and histograms too?

**Comment by fatsheep9146** [comment]
> > do we have tests that check that _created series are correctly parsed for summaries, histograms, and counters?
> 
> Okay, so double checked there are positive tests for _created when it's actually _created timestamp being parsed and used for all of the three. We have also a test case for a counter where _created is not the actual created timestamp and is being left as is. Should I add a test case for the latter for summaries and histograms too?

I think we'd better cover them all.

**Comment by dashpole** [comment]
Discussed at the prometheus wg meeting.  There are already tests for the _created suffix that work correctly.

**Review by dashpole — APPROVED** [review]

**Comment by jmichalek132** [comment]
Pipeline failure for a different component unrelated to my changes.

**Review by fatsheep9146 — APPROVED** [review]

**Pull request merged**